Explorer/Assets/DCL/ApplicationVersionGuard/LauncherRedirectionScreenController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Cysharp.Threading.Tasks;
+using MVC;
+using System.Threading;
+
+#if !UNITY_EDITOR
+using UnityEngine;
+#endif
+
+namespace DCL.AuthenticationScreenFlow
+{
+    public class LauncherRedirectionScreenController : ControllerBase<LauncherRedirectionScreenView>
+    {
+        private readonly ApplicationVersionGuard.ApplicationVersionGuard versionGuard;
+        private readonly string current;
+        private readonly string latest;
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Overlay;
+
+        public LauncherRedirectionScreenController(ApplicationVersionGuard.ApplicationVersionGuard versionGuard, ViewFactoryMethod viewFactory, string current, string latest) : base(viewFactory)
+        {
+            this.versionGuard = versionGuard;
+            this.current = current;
+            this.latest = latest;
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            base.OnViewInstantiated();
+
+            viewInstance.SetVersions(current, latest);
+            viewInstance.CloseButton.onClick.AddListener(Quit);
+            viewInstance.CloseWithLauncherButton.onClick.AddListener(HandleVersionUpdate);
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            viewInstance.CloseButton.onClick.RemoveListener(Quit);
+            viewInstance.CloseWithLauncherButton.onClick.RemoveListener(HandleVersionUpdate);
+        }
+
+        private void HandleVersionUpdate()
+        {
+            versionGuard.LaunchOrDownloadLauncherAsync().Forget();
+        }
+
+        private static void Quit()
+        {
+#if UNITY_EDITOR
```

Comment:
The content of the quit function has been used in different places, maybe we can have an util and call it everywhere we need to stop/quit the application

Reply:
I was thinking the same... Maybe there is a good place somewhere